history/src/main/scala/org/apache/spark/deploy/history/SimpleFsHistoryProvider.scala

```diff
@@ -0,0 +1,428 @@
+
+
+/**
+ * A class that provides application history from event logs stored in the file system.
+ * This provider will render the history application UI of specifically requested
+ * applications only.  No scanning or caching of event logs from a filesystem is performed.
+ *
+ */
+
+private[history] class SimpleFsHistoryProvider(conf: SparkConf, clock: Clock)
+  extends ApplicationHistoryProvider with Logging {
+
+  def this(conf: SparkConf) = {
+    this(conf, new SystemClock())
+  }
+  
+  import SimpleFsHistoryProvider._
+  
+  type ReplayEventsFilter = (String) => Boolean
+  val SELECT_ALL_FILTER: ReplayEventsFilter = { (eventString: String) => true }
+
+  private val logDir = conf.getOption("spark.history.fs.logDirectory")
+    .getOrElse(DEFAULT_LOG_DIR)
+
+  private val HISTORY_UI_ACLS_ENABLE = conf.getBoolean("spark.history.ui.acls.enable", false)
+  private val HISTORY_UI_ADMIN_ACLS = conf.get("spark.history.ui.admin.acls", "")
+  private val HISTORY_UI_ADMIN_ACLS_GROUPS = conf.get("spark.history.ui.admin.acls.groups", "")
+  logInfo(s"History server ui acls " + (if (HISTORY_UI_ACLS_ENABLE) "enabled" else "disabled") +
+    "; users with admin permissions: " + HISTORY_UI_ADMIN_ACLS.toString +
+    "; groups with admin permissions" + HISTORY_UI_ADMIN_ACLS_GROUPS.toString)
+
+  private val hadoopConf = SparkHadoopUtil.get.newConfiguration(conf)
+  private val fs = new Path(logDir).getFileSystem(hadoopConf)
+
+  private[history] def initialize(): Unit = {}
+  
+  override def getEventLogsUnderProcess(): Int = 0
+
+  override def getLastUpdatedTime(): Long = 0
+  
+  override def getListing(): Iterator[ApplicationInfo] = { null }
+  
+  override def onUIDetached(appId: String, attemptId: Option[String], ui: SparkUI): Unit = { }
+   
+  private val activeUIs = new mutable.HashMap[(String, Option[String]), LoadedAppUI]()
+  
+  def getApplicationAttemptInfo(fileStatus: FileStatus): Option[ApplicationAttemptInfo] = {
+ 
+    // When getting the Application Info, we really only want the key details - Name, ID, start time, stop time
+    val eventsFilter: ReplayEventsFilter = { eventString =>
+      eventString.startsWith(APPL_START_EVENT_PREFIX) ||
+      eventString.startsWith(APPL_END_EVENT_PREFIX)
+    }
+
+    // need to get the ApplicationAttemptInfo from the replay to return
+    val logPath = fileStatus.getPath()
+    val bus = new ReplayListenerBus()
+    val listener = new AppListingListener(fileStatus, clock)
+    bus.addListener(listener)
+    replay(fileStatus, bus, eventsFilter)
+
+    listener.applicationAttemptInfo
+  }
+  
+  def getApplicationInfoWrapper(appId: String): Option[ApplicationInfoWrapper] = {
+    val fileStatus = fs.getFileStatus(new Path(logDir, appId))
+    val logPath = fileStatus.getPath()
+    val bus = new ReplayListenerBus()
+    val listener = new AppListingListener(fileStatus, clock)
+    bus.addListener(listener)
+    replay(fileStatus, bus, SELECT_ALL_FILTER)
```

Comment:
To just get the ApplicationInfo we don't need to replay with *every* event, We can just filter by Start Time and End Time (like getApplicationAttemptInfo)